src/backend/cod1.c

```diff
@@ -4515,7 +4525,16 @@ code *loaddata(elem *e,regm_t *pretregs)
     }
     else if (sz <= REGSIZE)
     {
-        ce = loadea(e,&cs,0x8B,reg,0,RMload,0); // MOV reg,data
+        unsigned op = 0x8B;                     // MOV reg,data
+        if (sz == 2 && !I16 && config.target_cpu >= TARGET_PentiumPro &&
+            // Workaround for OSX linker bug:
+            //   ld: GOT load reloc does not point to a movq instruction in test42 for x86_64
+            !(config.exe & EX_OSX64 && !(sytab[e->EV.sp.Vsym->Sclass] & SCSS))
+           )
+        {
+            op = tyuns(tym) ? 0x0FB7 : 0x0FBF;  // MOVZX/MOVSX
```

Comment:
It's very challenging to write a straightforward code generator for x86. There's nothing consistent about anything, and every new member of the family tends to upset everything yet again.